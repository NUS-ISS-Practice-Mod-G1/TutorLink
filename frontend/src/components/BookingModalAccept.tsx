import React from "react";

interface BookingModalAcceptProps {
  booking: {
    studentName: string;
    date: Date;
    slot: { start: string; end: string };
    lessonType: string;
  };
  onClose: () => void;
  onAccept: () => void;
  onReject: () => void;
}

const BookingModalAccept: React.FC<BookingModalAcceptProps> = ({
  booking,
  onClose,
  onAccept,
  onReject,
}) => {
  return (
    <div className="fixed inset-0 flex items-center justify-center bg-black bg-opacity-40 z-50">
      <div className="bg-white rounded-xl shadow-lg p-6 w-96">
        <h2 className="text-lg font-semibold mb-4 text-gray-800">
          Accept Booking Request
        </h2>

        <p className="mb-1">👤 {booking.studentName}</p>
        <p className="mb-1">
          📅 {booking.date.toDateString()} | ⏰ {booking.slot.start} -{" "}
          {booking.slot.end}
        </p>
        <p className="mb-6">🎓 Lesson Type: {booking.lessonType}</p>

        <div className="flex justify-end gap-3 mb-3">
          <button
            onClick={onReject}
            className="px-4 py-2 rounded-md bg-red-600 text-white hover:bg-red-700 w-1/2"
          >
            Reject
          </button>
          <button
            onClick={onAccept}
            className="px-4 py-2 rounded-md bg-green-600 text-white hover:bg-green-700 w-1/2"
          >
            Accept
          </button>
        </div>

        {/* Cancel below */}
        <div className="flex justify-end">
          <button
            onClick={onClose}
            className="px-4 py-2 rounded-md bg-gray-200 hover:bg-gray-300"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  );
};

export default BookingModalAccept;
