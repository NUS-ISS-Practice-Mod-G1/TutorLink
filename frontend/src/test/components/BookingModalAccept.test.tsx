import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import BookingModalAccept from "@/components/BookingModalAccept";

describe("BookingModalAccept", () => {
  const mockOnClose = jest.fn();
  const mockOnAccept = jest.fn();
  const mockOnReject = jest.fn();

  const booking = {
    studentName: "Alice Tan",
    date: new Date("2025-11-05T10:00:00Z"),
    slot: { start: "10:00", end: "11:00" },
    lessonType: "Mathematics",
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  // ---------- BASIC RENDER ----------
  test("renders modal with correct booking info", () => {
    render(
      <BookingModalAccept
        booking={booking}
        onClose={mockOnClose}
        onAccept={mockOnAccept}
        onReject={mockOnReject}
      />
    );

    expect(screen.getByText("Accept Booking Request")).toBeInTheDocument();
    expect(screen.getByText(/Alice Tan/)).toBeInTheDocument();
    expect(screen.getByText(/Mathematics/)).toBeInTheDocument();
    expect(screen.getByText(/10:00 - 11:00/)).toBeInTheDocument();
    expect(screen.getByText(/Nov/)).toBeInTheDocument(); // formatted date
  });

  // ---------- BUTTONS EXIST ----------
  test("renders Accept, Reject, and Close buttons", () => {
    render(
      <BookingModalAccept
        booking={booking}
        onClose={mockOnClose}
        onAccept={mockOnAccept}
        onReject={mockOnReject}
      />
    );

    expect(screen.getByText("Accept")).toBeInTheDocument();
    expect(screen.getByText("Reject")).toBeInTheDocument();
    expect(screen.getByText("Close")).toBeInTheDocument();
  });

  // ---------- ACCEPT HANDLER ----------
  test("calls onAccept when Accept button clicked", () => {
    render(
      <BookingModalAccept
        booking={booking}
        onClose={mockOnClose}
        onAccept={mockOnAccept}
        onReject={mockOnReject}
      />
    );

    fireEvent.click(screen.getByText("Accept"));
    expect(mockOnAccept).toHaveBeenCalledTimes(1);
  });

  // ---------- REJECT HANDLER ----------
  test("calls onReject when Reject button clicked", () => {
    render(
      <BookingModalAccept
        booking={booking}
        onClose={mockOnClose}
        onAccept={mockOnAccept}
        onReject={mockOnReject}
      />
    );

    fireEvent.click(screen.getByText("Reject"));
    expect(mockOnReject).toHaveBeenCalledTimes(1);
  });

  // ---------- CLOSE HANDLER ----------
  test("calls onClose when Close button clicked", () => {
    render(
      <BookingModalAccept
        booking={booking}
        onClose={mockOnClose}
        onAccept={mockOnAccept}
        onReject={mockOnReject}
      />
    );

    fireEvent.click(screen.getByText("Close"));
    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });

  // ---------- SNAPSHOT ----------
  test("matches snapshot", () => {
    const { container } = render(
      <BookingModalAccept
        booking={booking}
        onClose={mockOnClose}
        onAccept={mockOnAccept}
        onReject={mockOnReject}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
