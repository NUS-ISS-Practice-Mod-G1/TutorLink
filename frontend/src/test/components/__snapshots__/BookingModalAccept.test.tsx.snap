// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BookingModalAccept matches snapshot 1`] = `
<div>
  <div
    class="fixed inset-0 flex items-center justify-center bg-black bg-opacity-40 z-50"
  >
    <div
      class="bg-white rounded-xl shadow-lg p-6 w-96"
    >
      <h2
        class="text-lg font-semibold mb-4 text-gray-800"
      >
        Accept Booking Request
      </h2>
      <p
        class="mb-1"
      >
        👤 
        Alice Tan
      </p>
      <p
        class="mb-1"
      >
        📅 
        Wed Nov 05 2025
         | ⏰ 
        10:00
         -
         
        11:00
      </p>
      <p
        class="mb-6"
      >
        🎓 Lesson Type: 
        Mathematics
      </p>
      <div
        class="flex justify-end gap-3 mb-3"
      >
        <button
          class="px-4 py-2 rounded-md bg-red-600 text-white hover:bg-red-700 w-1/2"
        >
          Reject
        </button>
        <button
          class="px-4 py-2 rounded-md bg-green-600 text-white hover:bg-green-700 w-1/2"
        >
          Accept
        </button>
      </div>
      <div
        class="flex justify-end"
      >
        <button
          class="px-4 py-2 rounded-md bg-gray-200 hover:bg-gray-300"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
